ArduinoNTPd/SerialDataSource.h

```diff
@@ -30,24 +30,52 @@ class SerialDataSource : public IDataSource
     /*
      * Starts serial comms.
      */
-    void begin()
+    void begin(uint16_t BAUD = 4800)
     {
-        #ifdef HARDWARE_SERIAL_CLASS
+#ifdef HARDWARE_SERIAL_CLASS
 #undef serialPort_
 #define serialPort_ HARDWARE_SERIAL_CLASS
 #endif
-
         // GPS module needs 4800 baud.
-        serialPort_.begin(4800);
-        
-        // Disable everything but $GPRMC
-        serialPort_.write("$PSRF103,00,00,01*08\r\n");
-        serialPort_.write("$PSRF103,01,00,01*09\r\n");
-        serialPort_.write("$PSRF103,02,00,01*0A\r\n");
-        serialPort_.write("$PSRF103,03,00,01*0B\r\n");
-        serialPort_.write("$PSRF103,04,00,01*0C\r\n");
-        serialPort_.write("$PSRF103,05,00,01*0D\r\n");
-        serialPort_.write("$PSRF103,08,00,01,00*2C\r\n");
+        serialPort_.begin(BAUD);
+
+        // Disable everything but RMC
+#if (GPS_MODULE == 0) // SiRF
+        Serial.println(F("SiRF"));
+        serialPort_.println(F("$PSRF103,00,00,01*08"));
+        serialPort_.println(F("$PSRF103,01,00,01*09"));
+        serialPort_.println(F("$PSRF103,02,00,01*0A"));
+        serialPort_.println(F("$PSRF103,03,00,01*0B"));
+        serialPort_.println(F("$PSRF103,04,00,01*0C"));
+        serialPort_.println(F("$PSRF103,05,00,01*0D"));
+        serialPort_.println(F("$PSRF103,08,00,01,00*2C"));
+
+#elif (GPS_MODULE == 1) // MediaTek (MTK)
+        Serial.println(F("MediaTek"));
+        serialPort_.println(F("$PMTK314,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0*29"));
+        // 1s NMEA update rate
+        serialPort_.println(F("$PMTK220,1000*1F")); // 1 Hz
+        //serialPort_.println(F("$PMTK220,200*2C")); // 5 Hz
```

Comment:
Remove commented code, or wrap in #ifdef if it's useful to use something other than 1Hz for the update rate.